packages/docusaurus-plugin-content-docs/src/docs.ts

```diff
@@ -365,6 +374,10 @@ export function addDocNavigation(
           `Error when loading ${doc.id} in ${doc.sourceDirName}: the pagination_${type} front matter points to a non-existent ID ${docId}.`,
         );
       }
+      // Gracefully handle explicitly providing an unlisted doc ID in production
```

Comment:
In which case we'd want to explicitly link to an unlisted doc? I'm not 100% sure it makes sense and I'd rather throw to prevent errors until someone comes with a good use-case to relax this rule

Do we have a unit test covering this edge case?

Reply:
Seems good enough

Unfortunately we don't have a good way to test it (yet) => will make sure to inject current env through docusaurus context